### ISSUE TITLE
env vars CAMUNDA_OPERATE_ELASTICSEARCH_NUMBEROFSHARDS and CAMUNDA_OPERATE_ELASTICSEARCH_NUMBEROFREPLICAS have no effect

### ISSUE THREAD

**Issue by igor-msk** [camunda/camunda#21205]
### Describe the bug

env vars CAMUNDA_OPERATE_ELASTICSEARCH_NUMBEROFSHARDS and CAMUNDA_OPERATE_ELASTICSEARCH_NUMBEROFREPLICAS have no effect - it stays with default values of 1 shard | 0 replicas

### To Reproduce
 ```
operate:
  env:
    - name: CAMUNDA_OPERATE_ELASTICSEARCH_NUMBEROFSHARDS
      value: "3"
    - name: CAMUNDA_OPERATE_ELASTICSEARCH_NUMBEROFREPLICAS
      value: "1"
```
### Current behavior
![image](https://github.com/user-attachments/assets/8dc935fc-b7d7-498a-a738-f6c9128338c9)

### Expected behavior

I get the required number of shards|replicas of operate indexes in ES

### Environment

- Operate Version: 8.5.5 and 8.5.4

---------------------------------------------------------------------------------------------

### Definition of Ready - Checklist

- [ ] The bug has been reproduced by the assignee in the environment compatible with the provided one; otherwise, the issue is closed with a comment
- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] The issue has been labeled with an appropriate `Bug-area` label
- [ ] Necessary screenshots, screen recordings, or files are attached to the bug report

For UI changes required to solve the bug:

- [ ] Design input has been collected by the assignee

### :point_right: Handover Dev to QA

- Changed components:
- Side effects on other components:
- Handy resources:
  BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc :
  
- Example projects:

- Commands/Steps needed to test; Versions to validate:

- Docker file / HELM chart : in case that it needed to be tested via docker share the version contain the fixed along with version of other services .

- Release version ( in which version this fixed/feature will be released):

**Comment by ThorbenLindhauer** [comment]
Hi @igor-msk,

do you observe this when the application creates the schema from scratch, or when you restart the application with updated shard/replica values when the indexes are already in place?

Cheers,
Thorben

**Comment by ThorbenLindhauer** [comment]
I assume that you want these settings to apply when you reconfigure Operate and restart it. This is currently not supported (see [docs](https://docs.camunda.io/docs/self-managed/operate-deployment/operate-configuration/#settings-for-shards-and-replicas)), but we have a [pending feature request](https://github.com/camunda/operate/issues/5361). I will close this ticket as a duplicate.

**Comment by igor-msk** [comment]
no, it happens when the application creates the schema from scratch

**Comment by igor-msk** [comment]
please reopen the bug

**Comment by ThorbenLindhauer** [comment]
Thanks for getting back to us.

Via the Helm chart, the Operate schema is created in an init container called `migration`. It sets up the schema or migrates it, and has a separate environment variable configuration as this:

```
operate:
  migration:
    env:
      - name: CAMUNDA_OPERATE_ELASTICSEARCH_NUMBEROFSHARDS
        value: "3"
      - name: CAMUNDA_OPERATE_ELASTICSEARCH_NUMBEROFREPLICAS
        value: "1"
```

Note that this configuration will fail with the latest Helm chart version 10.3.2 due to a bug in the template, but will be fixed with the upcoming 10.4.0 release via https://github.com/camunda/camunda-platform-helm/pull/2238.

**Comment by ThorbenLindhauer** [comment]
A bit more context: Moving to the init container has been [only introduced with Helm chart 10.3.0](https://github.com/camunda/camunda-platform-helm/pull/2154). I assume that any previous version would still accept the variables for the Operate (non-init) container.
